resources/lib/kodi/library.py

```diff
@@ -373,29 +411,64 @@ def update_item(item_task, library_home):
     export_item(item_task, library_home)
 
 
-def _update_running():
-    update = g.ADDON.getSetting('update_running') or None
+def _export_all_new_episodes_running():
+    update = g.PERSISTENT_STORAGE.get('export_all_new_episodes_running', False)
     if update:
-        starttime = common.strp(update, '%Y-%m-%d %H:%M')
-        if (starttime + timedelta(hours=6)) <= datetime.now():
-            g.ADDON.setSetting('update_running', 'false')
+        start_time = common.strp(g.PERSISTENT_STORAGE.get('export_all_new_episodes_start_time'),
+                                 '%Y-%m-%d %H:%M')
+        if datetime.now() >= start_time + timedelta(hours=6):
+            g.PERSISTENT_STORAGE['export_all_new_episodes_running'] = False
             common.warn('Canceling previous library update: duration >6 hours')
         else:
-            common.debug('DB Update already running')
+            common.debug('Export all new episodes is already running')
             return True
     return False
 
 
-def update_library():
+def export_all_new_episodes():
+    """
+    Update the local Kodi library with new episodes of every exported shows
     """
-    Update the local Kodi library with new episodes of exported shows
+    if not _export_all_new_episodes_running():
+        common.log('Starting to export new episodes for all tv shows')
+        common.debug('library')
+        common.debug(g.library())
+        g.PERSISTENT_STORAGE['export_all_new_episodes_running'] = True
+        g.PERSISTENT_STORAGE['export_all_new_episodes_start_time'] = datetime.now()\
+            .strftime('%Y-%m-%d %H:%M')
+
+        for library_item in g.library().values():
+            common.debug('mediatype {}'.format(library_item['videoid'].mediatype))
+            if library_item['videoid'].mediatype == common.VideoId.SHOW:
+                export_new_episodes(library_item['videoid'], False)
+            # add some randomness
```

Comment:
non sense comment, use explicit comment info (same at commit description)
if it is used to not be banned from netflix server or anything else it should be explained

Reply:
Indeed